src/main/java/org/ibp/api/java/impl/middleware/name/validator/NameTypeValidator.java

```diff
@@ -0,0 +1,81 @@
+package org.ibp.api.java.impl.middleware.name.validator;
+
+import org.apache.commons.lang3.StringUtils;
+import org.generationcp.middleware.api.nametype.GermplasmNameTypeDTO;
+import org.generationcp.middleware.api.nametype.GermplasmNameTypeRequestDTO;
+import org.generationcp.middleware.api.nametype.GermplasmNameTypeService;
+import org.ibp.api.exception.ApiRequestValidationException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.springframework.validation.BindingResult;
+import org.springframework.validation.MapBindingResult;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+@Component
+public class NameTypeValidator {
+
+	private static final Integer CODE_OR_NAME_MAX_LENGTH = 50;
+	private static final Integer DESCRPTION_MAX_LENGTH = 255;
+	private BindingResult errors;
+
+	@Autowired
+	GermplasmNameTypeService germplasmNameTypeService;
+
+	public void validate(final GermplasmNameTypeRequestDTO germplasmNameTypeRequestDTO) {
+		this.errors = new MapBindingResult(new HashMap<>(), GermplasmNameTypeRequestDTO.class.getName());
+
+		if (StringUtils.isBlank(germplasmNameTypeRequestDTO.getCode())) {
+			this.errors.reject("name.type.field.invalid", new String[] {"code"}, "");
```

Comment:
Use a specific property for each field. With this implementation it is impossible to make it work in another language. This comment applies for all errors using name.type.field.invalid